tools/ci_build/github/azure-pipelines/templates/use-android-emulator.yml

```diff
@@ -0,0 +1,63 @@
+# Android Emulator helpers
+
+parameters:
+- name: create
+  type: boolean
+  default: false
+
+- name: start
+  type: boolean
+  default: false
+
+- name: stop
+  type: boolean
+  default: false
+
+steps:
+- ${{ if eq(parameters.create, true) }}:
+  - script: |
+      set -e -x
+      python3 tools/python/run_android_emulator.py \
+        --android-sdk-root $(ANDROID_SDK_ROOT) \
+        --create-avd --system-image "system-images;android-31;default;x86_64"
+    displayName: Create Android Emulator
+
+- ${{ if eq(parameters.start, true) }}:
+  - script: |
+      if test -f $(Build.BinariesDirectory)/emulator.pid; then
+        echo "Emulator PID file was not expected to exist but does and has pid:" \
+          `cat $(Build.BinariesDirectory)/emulator.pid`
+        exit 1
+      fi
+    displayName: Check emulator.pid does not exist
+
+  # Add -verbose to --emulator-extra-args to additional logging.
+  - script: |
+      set -e -x
+      python3 tools/python/run_android_emulator.py \
+        --android-sdk-root $(ANDROID_SDK_ROOT) \
+        --start --emulator-extra-args="-partition-size 2047" \
+        --emulator-pid-file $(Build.BinariesDirectory)/emulator.pid
+      echo "Emulator PID:"`cat $(Build.BinariesDirectory)/emulator.pid`
+    displayName: Start Android Emulator
+
+- ${{ if eq(parameters.stop, true) }}:
+  - script: |
+      set -e -x
+      python3 -m pip install psutil
+    displayName: Install psutil for emulator shutdown by run_android_emulator.py
+    condition: always()
+
+  - script: |
+      if test -f $(Build.BinariesDirectory)/emulator.pid; then
```

Comment:
adding the check for the pid file is good, hopefully that reduces the failure rate of this step. maybe we should still have `set -e` (or check the exit code otherwise) to detect if the python script fails.